core/artwork.go

```diff
@@ -176,14 +179,19 @@ func resizeImage(reader io.Reader, size int) (io.ReadCloser, error) {
 	return ioutil.NopCloser(buf), err
 }
 
-func readFromTag(path string) (io.ReadCloser, error) {
-	f, err := os.Open(path)
+func readFromTag(fsys fs.FS, path string) (io.ReadCloser, error) {
+	f, err := fsys.Open(path)
 	if err != nil {
 		return nil, err
 	}
 	defer f.Close()
 
-	m, err := tag.ReadFrom(f)
```

Comment:
Wouldn't this cause problem for _huge_ files?

Reply:
Yep I didnt found any better solution yet sadly. The best alternative is probably to create a tmp file and then use a teewriter

Reply:
Potentially yes. This is a limitation of the `tag` lib used to extract embedded images. I may replace it with TagLib in the future (when I fix the Unicode issue with it)

Reply:
Ops, just misread this. I thought you were talking about the `tag.ReadFrom()`, which has the limitation I was talking about.

We don't want to read the whole file to memory to pass it to the `tag.ReadFrom()` method, and we also don't want to copy the whole file to a tmp file, as we don't need all pieces of it to read the embedded image. 

We need a `ReadSeeker` implementation out of the `fs.FS` instance.... Not sure how to achieve this, maybe using `http.FS()`?

Reply:
I replaced the ReadAll call with a copy to a tmpFile for now

Reply:
The os.DirFS returns the normal os.File instances but with the fs.File interface. We could just require the fs.File to also implement ReadSeeker and throwing an error if it doesnt

Reply:
Yep, that's actually what `http.FS` does. The tmpFile implementation is nice, but unfortunately it is not an good solution for this case, as it will cause music files to be copied when requesting album art. This is already slow, and would be a PITA having to copy the whole file. And if the music library is hosted in a cloud provider (my case with GDrive+Rclone), this will be the worse!

Reply:
Yup, I agree. We'll have to hack a proper `ReadSeeker` it seems.